### PR TITLE
16 feature/profile form

### PR DESCRIPTION
**Pull request by hereyubaraj**

### CODE REVIEW

File: app/src/main/java/np/com/naxa/factsnepal/ProfileInformationActivity.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        if (!validMunicipality) {
+            etMunicipality.setError(getString(R.string.msg_invalid_input, "municipality"));
+        }
+
+
+        if (!validStreet) {
+            etStreet.setError(getString(R.string.msg_invalid_input, "street"));
+        }
+
+        return validDob && validWard && validDistrict && validProvince && validMunicipality && validStreet;
```
Comment (hereyubaraj) [inline app/src/main/java/np/com/naxa/factsnepal/ProfileInformationActivity.java]:
```
validDob = dob.toString().isEmpty() ? showError(dob, 'empty dob') : dob.toString() ;
.......
private void showError( View view, String error_message ) {
  // statement
}
```

File: app/src/main/java/np/com/naxa/factsnepal/ProfileInformationActivity.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+    }
+
+    private void makeUIObservable() {
+        Flowable<CharSequence> dobChangeObservable = RxTextView.textChanges(etDob).skip(1).toFlowable(BackpressureStrategy.LATEST);
+        Flowable<CharSequence> wardChangeObservable = RxTextView.textChanges(etWard).skip(1).toFlowable(BackpressureStrategy.LATEST);
+        Flowable<CharSequence> districtChangeObservable = RxTextView.textChanges(etDistrict).skip(1).toFlowable(BackpressureStrategy.LATEST);
+        Flowable<CharSequence> provinceChangeObservable = RxTextView.textChanges(etProvience).skip(1).toFlowable(BackpressureStrategy.LATEST);
+        Flowable<CharSequence> municipalityChangeObservable = RxTextView.textChanges(etMunicipality).skip(1).toFlowable(BackpressureStrategy.LATEST);
+        Flowable<CharSequence> streetChangeObservable = RxTextView.textChanges(etStreet).skip(1).toFlowable(BackpressureStrategy.LATEST);
```
Comment (hereyubaraj) [inline app/src/main/java/np/com/naxa/factsnepal/ProfileInformationActivity.java]: comment this line of code and can keep this as a reference for future use.

**Comment by konishon** [comment]
please review this pr @hereyubaraj

File: app/src/main/java/np/com/naxa/factsnepal/ProfileInformationActivity.java (hunk 2 above)
Comment (konishon) [reply app/src/main/java/np/com/naxa/factsnepal/ProfileInformationActivity.java]: I agree with you

File: app/src/main/java/np/com/naxa/factsnepal/UpdateProfileActivity.java [hunk 3]
```diff
@@ -122,7 +121,6 @@ public void onComplete() {
     private boolean isValidForm(CharSequence dob, CharSequence ward, CharSequence district, CharSequence province, CharSequence municipality, CharSequence street) {
 
         boolean validDob, validWard, validDistrict, validProvince, validMunicipality, validStreet;
-//        validDob = Integer.parseInt(dob.toString()) > 1980 || Integer.parseInt(dob.toString()) < 2018;
         validDob = !dob.toString().isEmpty();
```
Comment (hereyubaraj) [inline app/src/main/java/np/com/naxa/factsnepal/UpdateProfileActivity.java:124]:
```
validDob = dob.toString().isEmpty() ? showError() : dob.toString();
```

**Pull request merged**